test/spec_request.rb

```diff
@@ -578,6 +578,10 @@ def self.req(headers)
     req = make_request \
       Rack::MockRequest.env_for("/", "HTTP_HOST" => "localhost:81", "HTTP_X_FORWARDED_HOST" => "example.org", "HTTP_FORWARDED" => "host=example.com:9292", "SERVER_PORT" => "9393")
     req.host_with_port.must_equal "example.com:9292"
+
+    req = make_request \
+      Rack::MockRequest.env_for("/", "HTTP_HOST" => "http://example.org", "HTTP_X_FORWARDED_HOST" => "")
+    req.host_with_port.must_equal "http://example.org"
```

Comment:
Should `#host_with_port` return the scheme? (`http://`)? Isn't that unexpected?

Reply:
You're right, `host_with_port` should not return a schema. My test data was incorrect. I'll fix the test so that `HTTP_HOST` does not include the schema.

Reply:
I've fixed it.